draft-westerlund-tsvwg-dtls-over-sctp-bis.md

```diff
@@ -508,40 +508,93 @@ ULP:  Upper Layer Protocol
    KeyUpdate. See {{sec-Consideration}} for security considerations
    regarding rekeying.
 
+### DTLS 1.2 Considerations
+
+   Before sending during renegotiation a ClientHello message or ServerHello
+   message, the DTLS endpoint MUST ensure that all DTLS messages using the
+   previous epoch have been acknowledged by the SCTP peer in a non-revokable
+   way.
+
+   Prior to processing a received ClientHello message or ServerHello
+   message, all other received SCTP user messages that are buffered in the
+   SCTP layer and can be delivered to the DTLS layer MUST be read and
+   processed by DTLS.
+
+### DTLS 1.3 Considerations
+
+   Before sending a KeyUpdate message, the DTLS endpoint MUST ensure that
+   all DTLS messages have been acknowledged by the SCTP peer in a non-revokable
+   way.
+   After sending the KeyUpdate message, it stops sending DTLS messages until
+   the corresponding Ack message has been processed.
+
+   Prior to processing a received KeyUpdate message, all other received SCTP
+   user messages that are buffered in the SCTP layer and can be delivered to
+   the DTLS layer MUST be read and processed by DTLS.
+
 ## DTLS Epochs
 
-   In general, DTLS implementations SHOULD discard records from
-   earlier epochs, as described in Section 4.2.1 of
-   {{I-D.ietf-tls-dtls13}}. To avoid discarding messages, the
-   processing guidelines in Section 4.2.1 of DTLS 1.3
-   {{I-D.ietf-tls-dtls13}} or Section 4.1 or DTLS 1.2 {{RFC6347}}
-   should be followed.
+   In general, DTLS implementations SHOULD discard records from earlier epochs.
+   However, in the context of a reliable communication this is not appropriate.
+
+### DTLS 1.2 Considerations
+
+   The procedures of Section 4.1 of {{RFC6347}} MUST NOT be followed.
+   Instead, when currently using epoch n, for n > 1, DTLS packets from epoch
+   n - 1 and n MUST be processed.
+
+### DTLS 1.3 Considerations
+
+   The procedures of Section 4.2.1 of {{RFC6347}} are irrelevant.
+   When receiving DTLS packets using epoch n, no DTLS packets from earlier
+   epochs are received.
```

Comment:
This implies draining of protection records. Couldn't we avoid this.

Reply:
Right now, I don't know now.